packages/flutter/test/material/carousel_test.dart

```diff
@@ -0,0 +1,461 @@
+  });
+
+  testWidgets('Carousel items customization', (WidgetTester tester) async {
+    final GlobalKey key = GlobalKey();
+    final ThemeData theme = ThemeData();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: theme,
+        home: Scaffold(
+          body: Carousel(
+            padding: const EdgeInsets.all(20.0),
+            backgroundColor: Colors.amber,
+            elevation: 10.0,
+            shape: const StadiumBorder(),
+            overlayColor: WidgetStateProperty.resolveWith((Set<WidgetState> states) {
+              if (states.contains(WidgetState.pressed)) {
+                return Colors.yellow;
+              }
+              if (states.contains(WidgetState.hovered)) {
+                return Colors.red;
+              }
+              if (states.contains(WidgetState.focused)) {
+                return Colors.purple;
+              }
+              return null;
+            }),
+            itemExtent: 200,
+            children: List<Widget>.generate(10, (int index) {
+              if (index == 0) {
+                return Center(
+                  key: key,
+                  child: Center(child: Text('Item $index')),
+                );
+              }
+              return Center(child: Text('Item $index'));
+            }),
+          ),
+        ),
+      ),
+    );
+
+    final Finder carouselMaterial = find.descendant(
+      of: find.byType(Carousel),
+      matching: find.byType(Material),
+    ).first;
+
+    expect(tester.getSize(carouselMaterial).width, 200 - 20 - 20); // Padding is 20 on both side.
+    final Material material = tester.widget<Material>(carouselMaterial);
+    expect(material.color, Colors.amber);
+    expect(material.elevation, 10.0);
+    expect(material.shape, const StadiumBorder());
+
+    RenderObject inkFeatures = tester.allRenderObjects.firstWhere((RenderObject object) => object.runtimeType.toString() == '_RenderInkFeatures');
+
+    // On hovered.
+    final TestGesture gesture = await _pointGestureToCarouselItem(tester, key);
+    await tester.pumpAndSettle();
+    expect(inkFeatures, paints..rect(color: Colors.red.withOpacity(1.0)));
+
+    // On pressed.
+    await tester.pumpAndSettle();
+    await gesture.down(tester.getCenter(find.byKey(key)));
+    await tester.pumpAndSettle();
+    inkFeatures = tester.allRenderObjects.firstWhere((RenderObject object) => object.runtimeType.toString() == '_RenderInkFeatures');
+    expect(inkFeatures, paints..rect()..rect(color: Colors.yellow.withOpacity(1.0)));
+
+    await tester.pumpAndSettle();
+    await gesture.up();
+    await gesture.removePointer();
+
+    // On focused.
+    final Element inkWellElement = tester.element(find.descendant(of: carouselMaterial, matching: find.byType(InkWell)));
+    expect(inkWellElement.widget, isA<InkWell>());
+    final InkWell inkWell = inkWellElement.widget as InkWell;
+
+    const MaterialState state = MaterialState.focused;
+
+    // Check overlay color in focused state
```

Comment:
for consistency with the other comments:

```suggestion
    // Check overlay color in focused state.
```